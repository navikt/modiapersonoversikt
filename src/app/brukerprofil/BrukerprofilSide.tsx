import * as React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { Action } from 'history';
import { RouteComponentProps, withRouter } from 'react-router';
import { connect, Dispatch } from 'react-redux';

import { paths } from '../routes/routing';
import BrukerprofilForm from './BrukerprofilForm';
import { AppState } from '../../redux/reducers';
import { Person, PersonRespons } from '../../models/person/person';
import Innholdslaster from '../../components/Innholdslaster';
import { hentAllPersonData } from '../../redux/restReducers/personinformasjon';
import { VeilederRoller } from '../../models/veilederRoller';
import { getVeilederRoller } from '../../redux/restReducers/veilederRoller';
import { STATUS } from '../../redux/restReducers/utils';
import Sidetittel from 'nav-frontend-typografi/lib/sidetittel';
import { RestReducer } from '../../redux/restReducers/restReducer';

const BrukerprofilWrapper = styled.section`
  flex-grow: 1;
  margin: 2em auto 3em;
  max-width: 720px;
  width: 90%;
  display: flex;
  flex-flow: column nowrap;
  animation: ${props => props.theme.animation.fadeIn};
  > *:not(:first-child):not(:nth-child(2)) {
    background-color: white;
    border-radius: ${props => props.theme.borderRadius.layout};
    margin: 1em 0;
    padding: 2em;
  }
`;

const LinkWrapper = styled.div`
  margin-bottom: 1em;
`;

interface RoutingProps {
    fodselsnummer: string;
}

interface DispatchProps {
    hentPersonData: (fødselsnummer: string) => void;
    hentVeilederRoller: () => void;
}

interface OwnProps {
    fødselsnummer: string;
    personReducer: RestReducer<PersonRespons>;
    veilederRollerReducer: RestReducer<VeilederRoller>;
}

type Props = RouteComponentProps<RoutingProps> & OwnProps & DispatchProps ;

class BrukerprofilSide extends React.Component<Props> {

    componentDidMount() {
        if (this.props.personReducer.status === STATUS.NOT_STARTED) {
            this.props.hentPersonData(this.props.fødselsnummer);
        }

        if (this.props.veilederRollerReducer.status === STATUS.NOT_STARTED) {
            this.props.hentVeilederRoller();
        }
    }

    render() {
        return (
            <BrukerprofilWrapper>
                <Innholdslaster
                    avhengigheter={[this.props.personReducer, this.props.veilederRollerReducer]}
                >
                    <LinkWrapper>
                        <Link
                            className={'lenke'}
                            to={`${paths.personUri}/${this.props.fødselsnummer}`}
                        >
                            {'<'} Tilbake
                        </Link>
                    </LinkWrapper>
                    <Sidetittel>Endre brukerprofil</Sidetittel>
                    <BrukerprofilForm
                        person={this.props.personReducer.data as Person}
                        veilderRoller={this.props.veilederRollerReducer.data}
                    />
                </Innholdslaster>
            </BrukerprofilWrapper>
        );
    }

}

const mapStateToProps = (state: AppState, ownProps: RouteComponentProps<RoutingProps>): OwnProps => {
    return ({
        fødselsnummer: ownProps.match.params.fodselsnummer,
        personReducer: state.restEndepunkter.personinformasjon,
        veilederRollerReducer: state.restEndepunkter.veilederRoller
    });
};

function mapDispatchToProps(dispatch: Dispatch<Action>): DispatchProps {
    return {
        hentPersonData: (fødselsnummer: string) => hentAllPersonData(dispatch, fødselsnummer),
        hentVeilederRoller: () => dispatch(getVeilederRoller())
    };
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(BrukerprofilSide));
