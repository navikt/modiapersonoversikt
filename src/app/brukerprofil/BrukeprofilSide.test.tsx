import * as React from 'react';
import { mount } from 'enzyme';

import BrukerprofilSide from './BrukerprofilSide';
import TestProvider from '../../test/Testprovider';

test('Smoketeste brukerprofil', () => {
    const brukerprofil = mount(
        <TestProvider>
            <BrukerprofilSide />
        </TestProvider>
    );

    expect(brukerprofil).toExist();
});
