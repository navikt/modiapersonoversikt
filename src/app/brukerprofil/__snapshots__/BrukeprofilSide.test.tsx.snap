// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Smoketeste brukerprofil 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 > *:first-child {
  margin-bottom: 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="alertstripe alertstripe--info"
    >
      <span
        class="alertstripe__ikon"
      >
        <span
          class="sr-only"
        >
          info
        </span>
        <svg
          aria-label="info-ikon"
          focusable="false"
          height="1.5em"
          kind="info-sirkel-fyll"
          role="img"
          viewBox="0 0 24 24"
          width="1.5em"
        >
          <g
            fill="none"
          >
            <path
              d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
              fill="#337C9B"
            />
            <path
              d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
              fill="#FFF"
            />
          </g>
        </svg>
      </span>
      <div
        class="typo-normal alertstripe__tekst"
      >
        Redigering av brukerprofil er flyttet
      </div>
    </div>
    <a
      class="lenkepanel"
      href=""
    >
      GÃ¥ til den nye personforvalteren
      <span
        class="lenkepanel__indikator"
      />
    </a>
  </div>
</div>
`;
